swagger.yaml

```diff
@@ -1,42 +1,223 @@
+    description: "The request does not contain the correct parameters and/or values in its parameters"
+
 definitions:
-  HelloResponse:
+
+  Job:
+    type: object
+    description: "A search reindex job. The information it contains is to be used by the developer who creates the Job, which is to replace an existing elastic search index with a new one."
+    required:
+      - "id"
+      - "last_updated"
+      - "links"
+      - "search_index_name"
+    properties:
+      id:
+        type: string
+        description: "A unique identifier for this specific search reindex job."
+      last_updated:
+        type: string
+        format: date-time
+        description: "A ISO8601 timestamp representing the last time the resource was updated."
+        example: "2021-03-18T09:03:41+0000"
+      links:
+        type: object
+        description: "Links that can be used to get the details of the Job (self) or of its tasks."
+        required:
+          - "tasks"
+          - "self"
+        properties:
+          tasks:
+            type: string
+            format: http://localhost:<port>/jobs/{id}/tasks
+            example: http://localhost:12150/jobs/abc123/tasks
+          self:
+            type: string
+            format: http://localhost:<port>/jobs/{id}
+            example: http://localhost:12150/jobs/abc123
+      number_of_tasks:
+        type: integer
+        description: "The number of tasks that are part of this search reindex job."
+      reindex_completed:
+        type: string
+        format: date-time
+        description: "A ISO8601 timestamp representing the date and time when the reindex job was completed."
+        example: "2021-03-18T09:03:41+0000"
+      reindex_failed:
+        type: string
+        format: date-time
+        description: "A ISO8601 timestamp representing the date and time when the reindex job failed."
+        example: "2021-03-18T09:03:41+0000"
+      reindex_started:
+        type: string
+        format: date-time
+        description: "A ISO8601 timestamp representing the date and time when the reindex job was started."
+        example: "2021-03-18T09:03:41+0000"
+      search_index_name:
+        type: string
+        description: "The name of the search index to be recreated by this search reindex job."
+      state:
+        type: string
+        description: "The current state of this search reindex job."
+        example: "created"
+        enum:
+          - created
+          - in-progress
+          - completed
+          - failed
+      total_search_documents:
+        type: integer
+        description: "The total number of documents that are held within the search index in question."
+      total_inserted_search_documents:
+        type: integer
+        description: "The total number of documents inserted into the new search index for a reindex job."
+
+  Jobs:
     type: object
+    description: "A list of Job resources held in an array."
+    required:
+      - "count"
```

Comment:
Discussed with Nathan and -limit, offset, sort  is mandatory , so it needs to be added to the list.

Reply:
ok, done